File: Maximum-Depth-of-Binary-Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

 // using BFS
public class Solution {
    public int MaxDepth(TreeNode root) {
        if(root == null){
            return 0 ;
        }
        int level = 0 ;
        Queue<TreeNode> q = new Queue<TreeNode>();
        q.Enqueue(root);

        while(q.Count > 0 ){
            int size = q.Count;
            for(int i = 0 ;i<size;i++){
                TreeNode cur = q.Dequeue();
                if(cur.left != null){
                    q.Enqueue(cur.left);
                }
                if(cur.right !=null){
                    q.Enqueue(cur.right);
                }
            }
            level += 1 ; 
        }
        return level;
    }
}

File: Maximum-Depth-of-Binary-Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
 // using iterative DFS
var maxDepth = function(root) {
    if(root == null){
        return 0 ;
    }

    let stack = [[root,1]];
    let max = 0;

    while (stack.length > 0) {
        const [node, depth] =  stack.pop();

        if (node != null) {
            max = Math.max(max, depth);

            if (node.left) stack.push([node.left, depth + 1]);
            if (node.right) stack.push([node.right, depth + 1]);

        }
    }
    return max ; 

};